packages/wrapped-keys-lit-actions/src/lib/internal/ethereum/signTransaction.ts

```diff
@@ -26,73 +38,107 @@ export function getValidatedUnsignedTx(unsignedTransaction) {
       chainId: unsignedTransaction.chainId,
       data: unsignedTransaction.dataHex,
     };
-  } catch (err) {
-    throw new Error(`Invalid unsignedTransaction - ${err.message}`);
+  } catch (err: unknown) {
+    throw new Error(`Invalid unsignedTransaction - ${(err as Error).message}`);
   }
 }
 
-async function getLatestNonce({ walletAddress, chain }) {
+async function getLatestNonce({
+  walletAddress,
+  chain,
+}: {
+  walletAddress: string;
+  chain: string;
+}) {
   try {
     const nonce = await Lit.Actions.getLatestNonce({
       address: walletAddress,
       chain: chain,
     });
 
     return nonce;
-  } catch (err) {
-    throw new Error(`Unable to get latest nonce - ${err.message}`);
+  } catch (err: unknown) {
+    throw new Error(`Unable to get latest nonce - ${(err as Error).message}`);
   }
 }
 
-async function getEthersRPCProvider({ chain }) {
+async function getEthersRPCProvider({ chain }: { chain: string }) {
   try {
     const rpcUrl = await Lit.Actions.getRpcUrl({
       chain,
     });
 
     return new ethers.providers.JsonRpcProvider(rpcUrl);
-  } catch (err) {
-    throw new Error(`Getting the rpc for the chain: ${chain} - ${err.message}`);
+  } catch (err: unknown) {
+    throw new Error(
+      `Getting the rpc for the chain: ${chain} - ${(err as Error).message}`
+    );
   }
 }
 
-async function getGasPrice({ userProvidedGasPrice, provider }) {
+async function getGasPrice({
+  userProvidedGasPrice,
+  provider,
+}: {
+  userProvidedGasPrice?: string;
+  provider: any;
```

Comment:
yea those i made it as `any` is because it doesn't like it as a type, it only works when use is it as a `const`
<img width="367" alt="image" src="https://github.com/user-attachments/assets/7337602e-c260-4b2c-8a74-134841355f57">

maybe it shouldn't be declared as a `const` in the global? 

<img width="443" alt="image" src="https://github.com/user-attachments/assets/715ff0b3-c7ef-41c0-8698-f54f2d2c5aa5">

Reply:
I think adding `import { ethers } from 'ethers';` and adding `ethers` as `external` in esbuild should do the trick without bundling it

Reply:
It looks like it works! - i'm thinking now if we should include this line though, cus it implies that we are also bundling the package, unless the user check the esBuild config `external` property themselves.

Reply:
actually we can just put a comment on the import